apps/src/StudioApp.js

```diff
@@ -130,6 +138,18 @@ var StudioApp = function () {
   this.maxRequiredBlocksToFlag_ = 1;
 
   /**
+  * @type {!TestableBlock[]}
+  */
+  this.recommendedBlocks_ = [];
+
+  /**
+  * The number of recommended blocks to give hints about at any one time.
+  * Set this to Infinity to show all.
+  * @type {!number=}
```

Comment:
Nit: I'm not sure this JSDoc annotation is valid -- how can it be non-nullable and also optional?